examples/ota-provider-app/ota-provider-common/BdxOtaSender.cpp

```diff
@@ -82,13 +82,18 @@ void BdxOtaSender::HandleTransferSessionOutput(TransferSession::OutputEvent & ev
             // end of the transfer.
             sendFlags.Set(chip::Messaging::SendMessageFlags::kExpectResponse);
         }
-        VerifyOrReturn(mExchangeCtx != nullptr, ChipLogError(BDX, "%s: mExchangeCtx is null", __FUNCTION__));
+        VerifyOrReturn(mExchangeCtx != nullptr, ChipLogError(BDX, "mExchangeCtx is null"));
         err = mExchangeCtx->SendMessage(event.msgTypeData.ProtocolId, event.msgTypeData.MessageType, std::move(event.MsgData),
                                         sendFlags);
         if (err != CHIP_NO_ERROR)
         {
             ChipLogError(BDX, "SendMessage failed: %s", chip::ErrorStr(err));
         }
+        if (event.msgTypeData.HasMessageType(chip::Protocols::SecureChannel::MsgType::StatusReport))
```

Comment:
```suggestion
        if (err == CHIP_NO_ERROR && !sendFlags.Has(chip::Messaging::SendMessageFlags::kExpectResponse))
```
would be better, in case someone changes the conditions under which that flag gets set or not set above.

Reply:
Incorporated this change locally